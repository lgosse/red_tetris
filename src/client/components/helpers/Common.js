import React from "react";

import styled from "styled-components";
import global from "../../styles/global";

export const Spacer = styled.div`
  flex: 1;
`;

export const Button = styled.button`
  padding: 15px 25px;
  font-family: ${global.font.family.game};
  transition: all 0.5s;
  border: none;
  background-color: unset;
  outline: none;
  cursor: pointer;

  ${props => props.width !== undefined && `width: ${props.width}`};
  ${props => props.margin !== undefined && `margin: ${props.margin}`};
  ${props => props.marginTop !== undefined && `margin-top: ${props.marginTop}`};
  ${props =>
    props.marginBottom !== undefined && `margin-bottom: ${props.marginBottom}`};
  ${props =>
    props.marginLeft !== undefined && `margin-left: ${props.marginLeft}`};
  ${props =>
    props.marginRight !== undefined && `margin-right: ${props.marginRight}`};
  ${props =>
    props.primary
      ? `
          background-color: ${global.color.primary};
          border: 1px solid ${global.color.accent};
          color: ${global.color.accent};

          &:hover {
            background-color: ${global.color.accent};
            color: ${global.color.primary};
          }
        `
      : `
          background-color: ${global.color.accent};
          color: ${global.color.primary};

          &:hover {
            background-color: ${global.color.accent};
            color: ${global.color.accent};
          }
    `};
`;

export const FlexContainer = styled.div`
  display: flex;
  ${props => props.flex && `flex: 1`};
`;

export const FlexSpacer = styled.div`
  flex: 1;
`;

export const FullSizeContainer = styled.div`
  width: 100%;
  ${props => props.padding && `padding: ${props.padding}`};
  ${props => props.flexContainer && `display: flex`};
  ${props => props.flex && `flex: 1`};
`;

const Hexagone = styled.span`
  display: block;
  position: relative;
  width: 20px;
  height: 11.54700538px;
  background-color: ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};

  margin: 6.66666667px 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);

  &:before,
  &:after {
    content: "";
    position: absolute;
    width: 0;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
  }

  &:before {
    bottom: 100%;
    border-bottom: 6.66666667px solid
      ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};
  }

  &:after {
    top: 100%;
    width: 0;
    border-top: 6.66666667px solid
      ${props => (props.primary ? `#bb3c2f` : `#dcddd8`)};
  }
`;

const HexaContainer = styled.div`
  height: 6px;
`;

export const Paragraph = styled.div`
  ${props => {
    if (props.width && props.center) {
      return `margin: auto; width: ${props.width}; text-align: center`;
    } else if (props.width && !props.center) {
      return `width: ${props.width}`;
    } else if (!props.width && props.center) {
      return `text-align: center`;
    }
  }};
  ${props => props.color && `color: ${global.color[props.color]}`};
  font-size: ${props => (props.size ? props.size : `14px`)};
  ${props => props.padding && `padding: ${props.size}`};
  ${props => props.gameFont && `font-family: ${global.font.family.game}`};
`;

export const RedContainer = FullSizeContainer.extend`
  background-color: ${global.color.primary};
  background-image: ${global.assets.backgroundRed};
`;

export const LightContainer = FullSizeContainer.extend`
  background-image: ${global.assets.backgroundLight};
  background-color: ${global.color.accent};
`;

export const GameFont = styled.span`
  font-family: ${global.font.family.game};
`;

export const Input = styled.input`
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  outline: none;
  display: block;
  width: 234px;
  padding: 7px;
  border: none;
  border-bottom: 1px solid #ddd;
  background: transparent;
  margin-bottom: 10px;
  font: 16px Arial, Helvetica, sans-serif;
  height: 45px;
  color: ${global.color.accent};

  &::placeholder {
    color: ${global.color.accent};
  }
`;

export const HexaSeparator = props => (
  <HexaContainer>
    <Hexagone primary={props.primary} />
  </HexaContainer>
);

const FaIcon = props => <i className={`fa fa-${props.className}`} />;

export const Icon = styled(FaIcon)`
  ${props => props.primary && `color: ${global.color.primary}`};
  ${props => props.accent && `color: ${global.color.accent}`};
  ${props => props.size && `font-size: ${props.size}`};
`;
