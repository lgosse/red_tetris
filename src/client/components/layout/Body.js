import styled from "styled-components";
import global from "../../styles/global";

const Body = styled.div`
  width: 100%;
  padding: ${global.padding.gt};
`;

export default Body;
