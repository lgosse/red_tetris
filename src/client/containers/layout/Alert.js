import React from "react";
import { connect } from "react-redux";
import styled, { keyframes } from "styled-components";

const AlertDisplay = styled.div`
  position: absolute;
  right: 20px;
  bottom: 20px;
  max-width: 300px;
  width: 100%;
  opacity: 0;
  padding: 20px;
  background-color: #333333;
  color: white;
  border-radius: 5px;
  transition: all 0.3s;

  &.show {
    opacity: 1;
  }
`;

const Alert = ({ alert }) =>
  alert.message ? (
    <AlertDisplay className="show">{alert.message}</AlertDisplay>
  ) : (
    <AlertDisplay />
  );

const mapStateToProps = state => ({
  alert: state.alert
});

export default connect(mapStateToProps)(Alert);
