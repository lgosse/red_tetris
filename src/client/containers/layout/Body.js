import styled from "styled-components";
import global from "../../styles/global";

const Body = styled.div`
  width: 100%;
  background-color: ${global.color.primary};
  background-image: ${global.assets.backgroundRed};
  min-height: calc(100vh - 111px);
`;

export default Body;
