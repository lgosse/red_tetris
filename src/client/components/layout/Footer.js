import React from "react";
import styled from "styled-components";
import global from "../../styles/global";

import { FlexSpacer, Paragraph, Icon } from "../helpers/Common";

const Bar = styled.div`
  width: 100%;
  display: flex;
  padding: ${global.padding.md};
  background-color: ${global.color.primary};
  color: ${global.color.accent};
  font-weight: ${global.font.weight.bold};
  border-top: 1px solid #bb3c2f;
  background-image: ${global.assets.backgroundRed};
`;

const Footer = () => {
  return (
    <Bar padding="20px">
      <FlexSpacer />
      <Paragraph size="16px" gameFont>
        Made with <Icon className="terminal" /> by lgosse
      </Paragraph>
    </Bar>
  );
};

export default Footer;
