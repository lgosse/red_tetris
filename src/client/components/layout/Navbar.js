import React from "react";
import styled from "styled-components";
import global from "../../styles/global";
import { connect } from "react-redux";
import { Spacer } from "../helpers/Common";

const Bar = styled.div`
  width: 100%;
  display: flex;
  padding: ${global.padding.md};
  background-color: ${global.color.primary};
  color: ${global.color.accent};
  font-weight: ${global.font.weight.bold};
  box-shadow: ${global.shadow.light};
`;

const Title = styled.span`
  font-size: ${global.font.size.title};
  font-family: ${global.font.family.game};
  text-shadow: ${global.font.shadow.heavy};
`;

const Link = styled.button`
  color: ${global.color.accent};
  font-size: ${global.font.size.subtitle};
  font-family: ${global.font.family.game};
  text-shadow: ${global.font.shadow.light};
  transition: all 0.5s;
  background-color: unset;
  border: none;
  outline: none;
  cursor: pointer;

  &:hover {
    text-shadow: ${global.font.shadow.heavy};
  }
`;

const Navbar = () => {
  return (
    <Bar>
      <Title>RED TETRIS</Title>
      <Spacer />
      <Link>
        <i className="fa fa-trophy" /> RANKING
      </Link>
      <Link>
        <i className="fa fa-plus" /> NEW GAME
      </Link>
    </Bar>
  );
};

export default Navbar;
