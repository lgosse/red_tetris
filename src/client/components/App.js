import React from "react";
import { connect } from "react-redux";
import { Switch, Route, withRouter } from "react-router-dom";
import Home from "../components/views/home/Home";
import Navbar from "../components/layout/Navbar";
import Body from "../components/layout/Body";
import Footer from "../components/layout/Footer";
import Ranking from "../components/views/Ranking";

const App = ({ alert, props }) => {
  return (
    <div>
      <Navbar />
      <Body>{props.children}</Body>
      <Footer />
    </div>
  );
};

const mapStateToAppProps = (state, props) => {
  return {
    alert: state.alert,
    props
  };
};

// Testing purposes exports
export { App, mapStateToAppProps };

export default withRouter(connect(mapStateToAppProps)(App));
