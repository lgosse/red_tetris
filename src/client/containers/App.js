import React from "react";
import { connect } from "react-redux";
import { Switch, Route, withRouter } from "react-router-dom";
import Home from "../containers/views/Home";
import Navbar from "../containers/layout/Navbar";
import Body from "../containers/layout/Body";
import Footer from "../containers/layout/Footer";
import Alert from "../containers/layout/Alert";
import Ranking from "../containers/views/Ranking";

const App = ({ alert, props }) => (
  <div>
    <Navbar />
    <Body>{props.children}</Body>
    <Footer />
    <Alert />
  </div>
);

const mapStateToAppProps = (state, props) => ({
  alert: state.alert,
  props
});

// Testing purposes exports
export { App, mapStateToAppProps };

export default withRouter(connect(mapStateToAppProps)(App));
