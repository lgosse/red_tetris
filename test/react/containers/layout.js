import chai from "chai";
import React from "react";
import { Link } from "react-router-dom";
import { shallow } from "enzyme";
import equalJSX from "chai-equal-jsx";

// Elements to render
import {
  Bar,
  Title,
  NavLink,
  Navbar,
  mapStateToNavbarProps
} from "../../../src/client/containers/layout/Navbar";
import { Spacer } from "../../../src/client/components/helpers/Common.js";
import Footer from "../../../src/client/containers/layout/Footer";

import { App, mapStateToAppProps } from "../../../src/client/containers/App";

import { Connect, StyledComponent } from "../../helpers/mockComponents";
import { Alert } from "../../../src/client/containers/layout/Alert";

chai.should();
chai.use(equalJSX);

describe("Navbar", () => {
  describe("mapStateToNavbarProps", () => {
    it("should return an empty object", () => {
      const navbarProps = mapStateToNavbarProps({
        test: "this property should be ignored"
      });
      navbarProps.should.be.a("object").and.to.deep.equal({});
    });
  });

  describe("Component rendering", () => {
    it("should render as expected", () => {
      const output = shallow(<Navbar />);
      output.should.matchSnapshot();
    });
  });
});

describe("App", () => {
  describe("mapStateToAppProps", () => {
    it("should map alert to App props", () => {
      const state = {
        alert: {
          testProperty: "This property should be pass to App."
        }
      };
      const alert = "Props should be passed to App";
      const appProps = mapStateToAppProps(state, alert);

      appProps.should.be.deep.equal({
        alert: {
          testProperty: "This property should be pass to App."
        },
        props: "Props should be passed to App"
      });
    });
  });

  describe("Component rendering", () => {
    it("should render", () => {
      const state = {
        alert: {
          testProperty: "This property should be pass to App."
        }
      };
      const alert = "Props should be passed to App";
      const appProps = mapStateToAppProps(state, alert);

      const output = App(...appProps, <div />);
      output.should.matchSnapshot();
    });
  });
});

describe("Footer", () => {
  describe("Component rendering", () => {
    it("should render as expected", () => {
      const output = shallow(<Footer />);
      output.should.matchSnapshot();
    });
  });
});

describe("Alert", () => {
  describe("Component rendering", () => {
    it("should render as expected with message", () => {
      const alert = {
        message: "SASUUUKEEEE"
      };
      const output = shallow(<Alert alert={alert} />);
      output.should.matchSnapshot();
    });
    it("should render as expected without message", () => {
      const alert = {};
      const output = shallow(<Alert alert={alert} />);
      output.should.matchSnapshot();
    });
  });
});
