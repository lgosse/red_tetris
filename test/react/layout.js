import chai from "chai";
import React from "react";
import equalJSX from "chai-equal-jsx";
import { createRenderer } from "react-dom/test-utils";
import { Link } from "react-router-dom";

// Elements to render
import {
  Bar,
  Title,
  NavLink,
  Navbar,
  mapStateToNavbarProps
} from "../../src/client/containers/layout/Navbar";
import { Spacer } from "../../src/client/components/helpers/Common.js";

import { App, mapStateToAppProps } from "../../src/client/containers/App";

import { Connect, StyledComponent } from "../helpers/mockComponents";

chai.should();
chai.use(equalJSX);

describe("Navbar", () => {
  describe("mapStateToNavbarProps", () => {
    it("should return an empty object", () => {
      const navbarProps = mapStateToNavbarProps({
        test: "this property should be ignored"
      });
      navbarProps.should.be.a("object").and.to.deep.equal({});
    });
  });

  describe("Component rendering", () => {
    it("should render as expected", () => {
      const renderer = createRenderer();
      renderer.render(React.createElement(Navbar));
      const output = renderer.getRenderOutput();
      output.should.matchSnapshot();
    });
  });
});

describe("App", () => {
  describe("mapStateToAppProps", () => {
    it("should map alert to App props", () => {
      const state = {
        alert: {
          testProperty: "This property should be pass to App."
        }
      };
      const alert = "Props should be passed to App";
      const appProps = mapStateToAppProps(state, alert);

      appProps.should.be.deep.equal({
        alert: {
          testProperty: "This property should be pass to App."
        },
        props: "Props should be passed to App"
      });
    });
  });

  describe("Component rendering", () => {
    it("should render", () => {
      const state = {
        alert: {
          testProperty: "This property should be pass to App."
        }
      };
      const alert = "Props should be passed to App";
      const appProps = mapStateToAppProps(state, alert);

      const output = App(...appProps, <div />);
      output.should.matchSnapshot();
    });
  });
});
