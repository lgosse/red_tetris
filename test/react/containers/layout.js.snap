// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component rendering should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    alert={
      Object {
        "message": "SASUUUKEEEE",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "SASUUUKEEEE",
      "className": "show",
    },
    "ref": null,
    "rendered": "SASUUUKEEEE",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "SASUUUKEEEE",
        "className": "show",
      },
      "ref": null,
      "rendered": "SASUUUKEEEE",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Alert Component rendering should render as expected with message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    alert={
      Object {
        "message": "SASUUUKEEEE",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "SASUUUKEEEE",
      "className": "show",
    },
    "ref": null,
    "rendered": "SASUUUKEEEE",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "SASUUUKEEEE",
        "className": "show",
      },
      "ref": null,
      "rendered": "SASUUUKEEEE",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Alert Component rendering should render as expected without message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    alert={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`App Component rendering should render 1`] = `
<div>
  <Connect(Navbar) />
  <styled.div />
  <Footer />
  <Connect(Alert) />
</div>
`;

exports[`Footer Component rendering should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div />,
        <styled.div
          gameFont={true}
          size="16px"
        >
          Made with 
          <Styled(FaIcon)
            className="terminal"
          />
           by lgosse and tbayet
        </styled.div>,
      ],
      "padding": "20px",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Made with ",
            <Styled(FaIcon)
              className="terminal"
            />,
            " by lgosse and tbayet",
          ],
          "gameFont": true,
          "size": "16px",
        },
        "ref": null,
        "rendered": Array [
          "Made with ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "terminal",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " by lgosse and tbayet",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div />,
          <styled.div
            gameFont={true}
            size="16px"
          >
            Made with 
            <Styled(FaIcon)
              className="terminal"
            />
             by lgosse and tbayet
          </styled.div>,
        ],
        "padding": "20px",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Made with ",
              <Styled(FaIcon)
                className="terminal"
              />,
              " by lgosse and tbayet",
            ],
            "gameFont": true,
            "size": "16px",
          },
          "ref": null,
          "rendered": Array [
            "Made with ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "terminal",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " by lgosse and tbayet",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Navbar Component rendering should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.span>
          RED TETRIS
        </styled.span>,
        <styled.div />,
        <Link
          id="home-link"
          replace={false}
          to="/"
        >
          <styled.button>
            <i
              className="fa fa-home"
            />
             HOME
          </styled.button>
        </Link>,
        <Link
          id="ranking-link"
          replace={false}
          to="ranking"
        >
          <styled.button>
            <i
              className="fa fa-trophy"
            />
             RANKING
          </styled.button>
        </Link>,
        <Link
          id="new-game-link"
          replace={false}
          to="new-game"
        >
          <styled.button>
            <i
              className="fa fa-plus"
            />
             PLAY
          </styled.button>
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "RED TETRIS",
        },
        "ref": null,
        "rendered": "RED TETRIS",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.button>
            <i
              className="fa fa-home"
            />
             HOME
          </styled.button>,
          "id": "home-link",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-home"
              />,
              " HOME",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-home",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " HOME",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.button>
            <i
              className="fa fa-trophy"
            />
             RANKING
          </styled.button>,
          "id": "ranking-link",
          "replace": false,
          "to": "ranking",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-trophy"
              />,
              " RANKING",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-trophy",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " RANKING",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.button>
            <i
              className="fa fa-plus"
            />
             PLAY
          </styled.button>,
          "id": "new-game-link",
          "replace": false,
          "to": "new-game",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-plus"
              />,
              " PLAY",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-plus",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " PLAY",
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.span>
            RED TETRIS
          </styled.span>,
          <styled.div />,
          <Link
            id="home-link"
            replace={false}
            to="/"
          >
            <styled.button>
              <i
                className="fa fa-home"
              />
               HOME
            </styled.button>
          </Link>,
          <Link
            id="ranking-link"
            replace={false}
            to="ranking"
          >
            <styled.button>
              <i
                className="fa fa-trophy"
              />
               RANKING
            </styled.button>
          </Link>,
          <Link
            id="new-game-link"
            replace={false}
            to="new-game"
          >
            <styled.button>
              <i
                className="fa fa-plus"
              />
               PLAY
            </styled.button>
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "RED TETRIS",
          },
          "ref": null,
          "rendered": "RED TETRIS",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.button>
              <i
                className="fa fa-home"
              />
               HOME
            </styled.button>,
            "id": "home-link",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-home"
                />,
                " HOME",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-home",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " HOME",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.button>
              <i
                className="fa fa-trophy"
              />
               RANKING
            </styled.button>,
            "id": "ranking-link",
            "replace": false,
            "to": "ranking",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-trophy"
                />,
                " RANKING",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-trophy",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " RANKING",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.button>
              <i
                className="fa fa-plus"
              />
               PLAY
            </styled.button>,
            "id": "new-game-link",
            "replace": false,
            "to": "new-game",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-plus"
                />,
                " PLAY",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " PLAY",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
